okio/src/main/java/okio/Buffer.java

```diff
@@ -626,6 +627,81 @@ String readUtf8Line(long newline) throws EOFException {
     }
   }
 
+  @Override public int readUtf8CodePoint() throws EOFException {
+    if (size == 0) throw new EOFException();
+
+    byte b0 = getByte(0);
+    int codePoint;
+    int byteCount;
+    int min;
+
+    if ((b0 & 0x80) == 0) {
+      // 0xxxxxxx.
+      codePoint = b0 & 0x7f;
+      byteCount = 1; // 7 bits (ASCII).
+      min = 0x0;
+
+    } else if ((b0 & 0xe0) == 0xc0) {
+      // 0x110xxxxx
```

Comment:
Oooh, that's interesting. I might circle back on that. I'm going to leave it as-is for now though.